spec/01-unit/01-db/01-schema/01-schema_spec.lua

```diff
@@ -1804,6 +1804,73 @@ describe("schema", function()
       assert.falsy(err)
     end)
 
+    it("test mutually required checks specified by transformations", function()
+      local Test = Schema.new({
+        fields = {
+          { a1 = { type = "string" } },
+          { a2 = { type = "string" } },
+          { a3 = { type = "string" } },
+        },
+        transformations = {
+          { input = { "a2" },       func = function() return {} end },
+          { input = { "a1", "a3" }, func = function() return {} end },
+        }
+      })
+
+      local ok, err = Test:validate_update({
+        a1 = "foo"
+      })
+      assert.is_falsy(ok)
+      assert.match("all or none of these fields must be set: 'a1', 'a3'", err["@entity"][1])
```

Comment:
Could you confirm what happens in this case?

```
      local ok, err = Test:validate_update({
        a1 = "aaa",
        a2 = "bbb",
        a3 = "ccc",
        a4 = "ddd",
      }, {
        a1 = "foo"
      })
```

If I follow correctly, the first table is the full entity obtained via read-before-write and the second table is the input given by the user with `PATCH`, right?

This needs to fail. Imagine for example that `a1` is `algorithm` and `a3` is `password` and the transformation hashes the password. If this passes, it will leave `algorithm` and `password` inconsistent with each other. Just double-checking.

Reply:
It fails with:
```
all or none of these fields must be set: 'a1', 'a3'
```

Reply:
First table is a table that has fields from `process_auto_fields` and in case it has records, it also is merged with `rbw`. Second table is what user gave.